File: src/test/java/LincheckTest.java
import org.jetbrains.kotlinx.lincheck.LinChecker;
import org.jetbrains.kotlinx.lincheck.annotations.Operation;
import org.jetbrains.kotlinx.lincheck.annotations.Param;
import org.jetbrains.kotlinx.lincheck.paramgen.IntGen;
import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions;
import org.jetbrains.kotlinx.lincheck.strategy.stress.StressCTest;
import org.jetbrains.kotlinx.lincheck.strategy.stress.StressOptions;
import org.junit.Test;

@Param(name="value", gen = IntGen.class, conf = "1:3")
@StressCTest
public class LincheckTest {
    private SetImpl<Integer> set = new SetImpl<>();

    @Operation
    public void isEmpty() {
        set.isEmpty();
    }

    @Operation
    public void add(@Param(name = "value") Integer value) {
        set.add(value);
    }

    @Operation
    public void remove(@Param(name = "value") Integer value) {
        set.remove(value);
    }

    @Operation
    public void contains(@Param(name = "value") Integer value) {
        set.contains(value);
    }

    @Test
    public void stressTest() {
        LinChecker.check(LincheckTest.class);
    }
}

File: src/test/kotlin/LincheckTest.kt
import org.jetbrains.kotlinx.lincheck.LoggingLevel
import org.jetbrains.kotlinx.lincheck.annotations.Operation
import org.jetbrains.kotlinx.lincheck.annotations.Param
import org.jetbrains.kotlinx.lincheck.check
import org.jetbrains.kotlinx.lincheck.paramgen.IntGen
import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
import org.jetbrains.kotlinx.lincheck.strategy.stress.StressOptions
import org.junit.jupiter.api.Test

@Param(name = "value", gen = IntGen::class, conf = "1:3")
class LincheckTestKt {
    private val set = SetImpl<Int>()

    @Operation
    fun isEmpty() = set.isEmpty()

    @Operation
    fun add(@Param(name = "value") value: Int) = set.add(value)

    @Operation
    fun remove(@Param(name = "value") value: Int) = set.remove(value)

    @Operation
    fun contains(@Param(name = "value") value: Int) = set.contains(value)

    @Test
    fun stressTest() = StressOptions()
        .logLevel(LoggingLevel.INFO)
        .check(LincheckTestKt::class)

    @Test
    fun modelCheckingTest() = ModelCheckingOptions().check(this::class)
}